lib/shopify_app/session/session_repository.rb

```diff
@@ -2,21 +2,23 @@ module ShopifyApp
   class SessionRepository
     class ConfigurationError < StandardError; end
 
+    EXPECTED_METHODS = [:store, :retrieve].freeze
+    EXPECTED_SHOP_STORAGE_METHODS = (EXPECTED_METHODS + [:retrieve_by_shopify_domain]).freeze
+    EXPECTED_USER_STORAGE_METHODS = (EXPECTED_METHODS + [:retrieve_by_shopify_user_id]).freeze
+
     class << self
       def shop_storage=(storage)
         @shop_storage = storage
+        return unless storage
```

Comment:
Does `@shop_storage` need to be set to `nil` if `storage` is `nil`? If not, maybe swap lines 11 and 12.